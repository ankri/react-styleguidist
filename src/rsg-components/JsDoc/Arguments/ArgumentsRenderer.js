import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';
import Markdown from 'rsg-components/Markdown';
import Code from 'rsg-components/Code';
import Group from 'react-group';

const styles = ({ type, name }) => ({
	type: {
		fontSize: 'inherit',
		color: type,
	},
	name: {
		fontSize: 'inherit',
		color: name,
	},
	argumentsBlock: {
		marginTop: 6,
		fontSize: 'inherit',
	},
	argumentsHeader: {
		fontWeight: 'bold',
		fontSize: 'inherit',
	},
});

export function JsDocArguments({ classes, tags }) {
	function renderParams(params) {
		return (
			params.map((param) => {
				return (
					<Group key={`${param.name}`}>
						<Code className={classes.name}>{param.name}</Code>
						{param.type && ':'}
						{param.type && <Code className={classes.type}>{param.type.name}</Code>}
						<Markdown text={`â€” ${param.description}`} inline />
					</Group>
				);
			})
		);
	}

	return (
		<div className={classes.argumentsBlock}>
			<div className={classes.argumentsHeader}>Arguments</div>
			{ tags.param && renderParams(tags.param) }
			{ tags.arg && renderParams(tags.arg) }
			{ tags.argument && renderParams(tags.argument) }
		</div>
	);
}

JsDocArguments.propTypes = {
	classes: PropTypes.object.isRequired,
	tags: PropTypes.shape({
		arg: PropTypes.array,
		argument: PropTypes.array,
		param: PropTypes.array,
	}).isRequired,
};

export default Styled(styles)(JsDocArguments);
