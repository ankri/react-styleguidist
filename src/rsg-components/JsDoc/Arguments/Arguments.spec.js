import React from 'react';
import { ArgumentsRenderer } from './ArgumentsRenderer';

describe('Arguments', () => {
	it('renderer should render Arguments', () => {
		const testTags = {
			param: [{
				title: 'param',
				name: 'test',
				description: 'Test description',
				type: 'Test type',
			}],
		};

		const actual = render(
			<ArgumentsRenderer classes={{}} tags={testTags} />
		);

		expect(actual).toMatchSnapshot();
	});
});

