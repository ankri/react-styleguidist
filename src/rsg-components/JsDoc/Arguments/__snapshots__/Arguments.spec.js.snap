// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arguments renderer should render Arguments 1`] = `null`;
